__tests__/storage/storage.test.js

```diff
@@ -193,6 +195,35 @@ describe('config version', () => {
     storage.setApiVersion(null);
     expect(storage.getTokenDepositPercentage()).toEqual(TOKEN_DEPOSIT_PERCENTAGE);
   });
+
+  it('should get native token from version', async () => {
+    const store = new MemoryStore();
+    const storage = new Storage(store);
+    expect(storage.getNativeTokenData()).toEqual({ ...DEFAULT_NATIVE_TOKEN_CONFIG, uid: NATIVE_TOKEN_UID });
+    const version = { native_token: { name: 'Native', symbol: 'N' } };
+    storage.setApiVersion(version);
+    expect(storage.getNativeTokenData()).toEqual({ name: 'Native', symbol: 'N', uid: NATIVE_TOKEN_UID });
+    storage.setApiVersion(null);
+    expect(storage.getNativeTokenData()).toEqual({ ...DEFAULT_NATIVE_TOKEN_CONFIG, uid: NATIVE_TOKEN_UID });
+  });
+
+  it('should save HTR when no custom native token is on version', async () => {
+    const store = new MemoryStore();
+    const storage = new Storage(store);
+    await expect(storage.getToken(NATIVE_TOKEN_UID)).resolves.toEqual(null);
+    await storage.saveNativeToken();
```

Comment:
suggestion(non-blocking): The last nice to have test here would be to call `storage.saveNativeToken()` twice, to ensure its conditional is working properly.